torch/csrc/jit/codegen/cuda/manager.cpp

```diff
@@ -67,46 +88,21 @@ class CudaFusionManager {
     Canonicalize(graph, false);
     auto repr = graph->toString(false);
 
-    // create new graph_cache_ entry;
-    if (graph_cache_.count(repr) == 0) {
+    // create new graph_cache_ids_ entry if none existed yet;
+    if (graph_cache_ids_.count(repr) == 0) {
       int32_t kernel_id = getNextUniqueID();
-      graph_cache_[repr] = kernel_id;
+      graph_cache_ids_[repr] = kernel_id;
+      graph_cache_[kernel_id] = std::make_unique<GraphCache>(graph);
```

Comment:
`TORCH_CHECK(graph_cache_.insert({kernel_id, std::make_unique<GraphCache>(graph)}).second);`

This inserts _and_ checks that there was no previous entry for kernel_id